src/gui.rs

```diff
@@ -165,25 +153,39 @@ impl Application for State {
             Command::batch([
                 Command::perform(fetch_minecraft_versions(), Message::SetMcVersions),
                 Command::perform(fetch_loader_versions(), Message::SetLoaderVersions),
-            ])
+            ]),
         )
     }
 
     fn title(&self) -> String {
-        "Quilt Installer".to_owned()
+        "Quilt Installer".into()
     }
 
-    fn update(&mut self, message: Self::Message, _clipboard: &mut Clipboard) -> Command<Self::Message> {
+    fn update(&mut self, message: Self::Message) -> Command<Self::Message> {
         match message {
             Message::Interaction(interaction) => match interaction {
                 Interaction::ChangeClientLocation(location) => self.client_location = location,
                 Interaction::BrowseClientLocation => return Message::BrowseClientLocation.into(),
                 Interaction::Install => return Message::Install.into(),
-                Interaction::SelectInstallation(installation) => self.selected_installation = installation,
-                Interaction::SelectLoaderVersion(version) => self.selected_loader_version = Some(version),
-                Interaction::SelectMcVersion(version) => self.selected_minecraft_version = Some(version),
-                Interaction::EnableSnapshots(enable) => self.show_snapshots = enable,
-                Interaction::EnableBetas(enable) => self.show_betas = enable,
+                Interaction::SelectInstallation(i) => self.installation_type = i,
+                Interaction::SelectLoaderVersion(v) => self.selected_loader_version = Some(v),
+                Interaction::SelectMcVersion(v) => self.selected_minecraft_version = Some(v),
+                Interaction::ToggleSnapshots(enable) => {
+                    self.show_snapshots = enable;
+                    self.selected_minecraft_version = self
```

Comment:
Yes it does.
I think the only time anyone would use this checkbox in normal usage is to get access to the betas. If you wanted to access the beta, I would assume you wanted the latest version. I personally find this behaviour much better, and it mimics the current installer too.

Reply:
The assumption is that if you select beta, you want the latest beta. that might be reasonable.
I can think of a few scenarios where this would be very annoying.

I'm not too hung up on this, but maybe we can have @EnnuiL as a tiebreaker?

Reply:
It shouldn't be hard to just disabled the snapping once the user selects a version. I think that would solve both our issues.

Reply:
I thought about just adding the option "Latest", which would be the default could just change according to the current checkboxes. It's a bit of extra work, but possibly the ideal solution.

To clarify: By default it would show as "Latest(1.19.2)" and if you enable snapshots it automatically switches to "Latest(1.19.3-pre1)"

Reply:
Yeah that's fine too!

Reply:
> enable snapshots it automatically switches to "Latest(1.19.3-pre1)"

Does this mean that the latest snapshot/beta excludes the stable versions?

Reply:
no, right now it would show up as "Latest(1.19.3)" with and without snapshots enabled.

Reply:
Hm, iirc, on my own fork? I pretty much left the version alone *but* if you disabled snapshots while you were on one? It snapped back to the latest stable version